packages/flutter_tools/lib/src/project.dart

```diff
@@ -467,6 +467,25 @@ class IosProject extends FlutterProjectPlatform implements XcodeBasedProject {
     await _updateGeneratedXcodeConfigIfNeeded();
   }
 
+  /// Check if one the [targets] of the project is a watchOS companion app target.
+  Future<bool> containsWatchCompanion(List<String> targets) async {
```

Comment:
I took the liberty of writing some unit tests for this since I know this file is really gross to test.  You can put them in `project_test.dart` or I'd be happy to push it to your fork, if you prefer.
```dart
    group('watch companion', () {
      MemoryFileSystem fs;
      MockPlistUtils mockPlistUtils;
      MockXcodeProjectInterpreter mockXcodeProjectInterpreter;
      FlutterProjectFactory flutterProjectFactory;
      setUp(() {
        fs = MemoryFileSystem.test();
        mockPlistUtils = MockPlistUtils();
        mockXcodeProjectInterpreter = MockXcodeProjectInterpreter();
        flutterProjectFactory = FlutterProjectFactory();
      });

      testUsingContext('cannot find bundle identifier', () async {
        final FlutterProject project = await someProject();
        expect(await project.ios.containsWatchCompanion(<String>['WatchTarget']), isFalse);
      }, overrides: <Type, Generator>{
        FileSystem: () => fs,
        ProcessManager: () => FakeProcessManager.any(),
        PlistParser: () => mockPlistUtils,
        XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
        FlutterProjectFactory: () => flutterProjectFactory,
      });

      group('with bundle identifier', () {
        setUp(() {
          when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenAnswer(
              (_) {
              return Future<Map<String,String>>.value(<String, String>{
                'PRODUCT_BUNDLE_IDENTIFIER': 'io.flutter.someProject',
              });
            }
          );
        });

        testUsingContext('no Info.plist in target', () async {
          final FlutterProject project = await someProject();
          expect(await project.ios.containsWatchCompanion(<String>['WatchTarget']), isFalse);
        }, overrides: <Type, Generator>{
          FileSystem: () => fs,
          ProcessManager: () => FakeProcessManager.any(),
          PlistParser: () => mockPlistUtils,
          XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
          FlutterProjectFactory: () => flutterProjectFactory,
        });

        testUsingContext('Info.plist in target does not contain WKCompanionAppBundleIdentifier', () async {
          final FlutterProject project = await someProject();
          project.ios.hostAppRoot.childDirectory('WatchTarget').childFile('Info.plist').createSync(recursive: true);

          expect(await project.ios.containsWatchCompanion(<String>['WatchTarget']), isFalse);
        }, overrides: <Type, Generator>{
          FileSystem: () => fs,
          ProcessManager: () => FakeProcessManager.any(),
          PlistParser: () => mockPlistUtils,
          XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
          FlutterProjectFactory: () => flutterProjectFactory,
        });

        testUsingContext('target WKCompanionAppBundleIdentifier is not project bundle identifier', () async {
          final FlutterProject project = await someProject();
          project.ios.hostAppRoot.childDirectory('WatchTarget').childFile('Info.plist').createSync(recursive: true);

          when(mockPlistUtils.getValueFromFile(any, 'WKCompanionAppBundleIdentifier')).thenReturn('io.flutter.someOTHERproject');
          expect(await project.ios.containsWatchCompanion(<String>['WatchTarget']), isFalse);
        }, overrides: <Type, Generator>{
          FileSystem: () => fs,
          ProcessManager: () => FakeProcessManager.any(),
          PlistParser: () => mockPlistUtils,
          XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
          FlutterProjectFactory: () => flutterProjectFactory,
        });

        testUsingContext('has watch companion', () async {
          final FlutterProject project = await someProject();
          project.ios.hostAppRoot.childDirectory('WatchTarget').childFile('Info.plist').createSync(recursive: true);
          when(mockPlistUtils.getValueFromFile(any, 'WKCompanionAppBundleIdentifier')).thenReturn('io.flutter.someProject');

          expect(await project.ios.containsWatchCompanion(<String>['WatchTarget']), isTrue);
        }, overrides: <Type, Generator>{
          FileSystem: () => fs,
          ProcessManager: () => FakeProcessManager.any(),
          PlistParser: () => mockPlistUtils,
          XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
          FlutterProjectFactory: () => flutterProjectFactory,
        });
      });
    });
```

Reply:
Thanks a lot! Totally agree that tests for containsWatchCompanion should be included (... and that writing tests for the plist file is not exactly fun ;-) )

Reply:
> (... and that writing tests for the plist file is not exactly fun ;-) )

Ugh I know, tell me about it. 😝 
We're working on making tool tests easier to write, though mocking out a plist parser may never be anyone's favorite pastime.